docs/dev/checks/README.md

```diff
@@ -49,21 +47,80 @@ It's up to the check author how to structure configuration data.
 Each instances of a check are completely independent from one another and might
 run at different intervals.
 
-## Python Checks
+## Anatomy of a Python Check
+
+Same as any built-in integration, a Custom Check consists of a Python class that
+inherits from `AgentCheck` and implements the `check` method:
+
+```python
+from datadog_checks.checks import AgentCheck
+
+class MyCheck(AgentCheck):
+    def check(self, instance):
+        # Collect metrics, emit events, submit service checks,
+        # ...
+```
 
-### API
+The Agent creates an object of type `MyCheck` for each element contained in the
+`instances` sequence within the corresponding config file:
 
-Read more about the [Python Checks API](python/check_api.md).
+```
+instances:
+  - host: localhost
+    port: 6379
+
+  - host: example.com
+    port: 6379
+```
+
+Any mapping contained in `instances` is passed to the `check` method through the
+named parameter `instance`. The `check` method is invoked at every run of the
+[collector][collector].
+
+The `AgentCheck` base class provides several useful attributes and methods,
+refer to the [Python docs][datadog_checks_base] and the developer
+[documentation pages][developer_docs] for more details.
+
+### Running subprocesses
+
+Due to the Python interpreter being embedded in an inherently multi-threaded environment (the go runtime)
+there are some limitations to the way Python Checks can run subprocesses.
+
+To run a subprocess from your check, use the `get_subprocess_output` function
+provided in `datadog_checks.utils.subprocess_output`:
+
+```python
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+
+class MyCheck(AgentCheck):
+    def check(self, instance):
+    # [...]
+    out, err, retcode = get_subprocess_output(cmd, self.log, raise_on_empty_output=True)
+```
 
-### Built-in packages
+Using the `subprocess` and `multiprocessing` modules provided by the Python standard library is _not
+supported_, and may result in your Agent crashing and/or creating processes that remain in a stuck or zombie
+state.
 
-The Agent provides a set of python packages that are built-in and only available
-within the embedded CPython interpreter:
+### Custom built-in modules
 
-- [aggregator](python/aggregator.md)
-- [datadog-agent](python/datadog_agent.md)
+A set of Python modules is provided capable to interact with a running Agent at
```

Comment:
It's weird to document modules which should never be used directly and not document the functions in `AgentCheck` that should be used.

Should we document the usage through `AgentCheck` in each `builtings/*.md` or document them here and with a link to the C/C++ doc  ?
Also https://datadog-checks-base.readthedocs.io/en/latest/datadog_checks_dev.api.html#module-datadog_checks.dev.utils doesn't seems to document how to write a check at all (like how to get the agent version, submit metrics ...).

It feels like we're missing a piece of doc in between the two.

Reply:
These docs aren't rendered in our documentation site, their main function is to give more context to devs working on the Agent codebase, more than integrations developers. 

I'd keep the two separated to minimize maintenance efforts, you don't want to come here every time the integrations team changes something in the base class, and whomever is browsing these docs has a link to https://datadog-checks-base.readthedocs.io/.

Docs about "how to write a check" (that are part of our official docs) are linked as well, so we should be good.